core/vm/evm.go

```diff
@@ -450,16 +465,25 @@ func openDB() {
 func storeCode(address string, creationCode []byte, deployedCodeHash string) {
 	codeMu.Lock()
 	defer codeMu.Unlock()
+
 	_, err := codeDB.Exec(`
-		INSERT INTO creationCode(code) VALUES (?) ON CONFLICT DO NOTHING;
-		INSERT INTO codeHash(hash) VALUES (?) ON CONFLICT DO NOTHING;
-		INSERT OR REPLACE INTO main(address, creationCode, deployedCodeHash)
-			SELECT ?, creationCode.id, codeHash.id FROM creationCode, codeHash
-			WHERE creationCode.code = ? AND codeHash.hash = ?;`,
-		creationCode, deployedCodeHash, address, creationCode, deployedCodeHash)
+		INSERT INTO creationCode(code) VALUES ($1) ON CONFLICT DO NOTHING;`,
+		creationCode)
+	_, err = codeDB.Exec(`
+		INSERT INTO codeHash(hash) VALUES ($1) ON CONFLICT DO NOTHING;`,
+		deployedCodeHash)
+	_, err = codeDB.Exec(`
+		INSERT INTO main(address, creationCode, deployedCodeHash)
+			SELECT $1, creationCode.id, codeHash.id FROM creationCode, codeHash
+			WHERE creationCode.code = $2 AND codeHash.hash = $3;`,
+		address, creationCode, deployedCodeHash)
```

Comment:
This code used to be different - it re-used the newly created primary id on insert - is this not rather expensive to do a full comparison on every insert?